src/attached_probe.cpp

```diff
@@ -830,18 +827,23 @@ void AttachedProbe::load_prog(BPFfeature &feature)
         // Don't attempt to load the program if the BTF load failed.
         // This will fall back to the error handling for failed program load,
         // which is more robust.
-        if (btf_fd_ >= 0)
+        if (btf_fd >= 0)
+        {
           progfd_ = bpf_prog_load(static_cast<::bpf_prog_type>(prog_type),
                                   name.c_str(),
                                   license,
                                   reinterpret_cast<const struct bpf_insn *>(
                                       insns.data()),
                                   insns.size() / sizeof(struct bpf_insn),
                                   &opts);
+          close(btf_fd);
+        }
       }
 
       if (opts.attach_prog_fd > 0)
         close(opts.attach_prog_fd);
+      if (opts.attach_btf_obj_fd > 0)
+        close(opts.attach_btf_obj_fd);
```

Comment:
It seems like we were closing the wrong fd here. Can we remove the closing of `opts.attach_prog_fd` as we don't seem to set that anywhere?

Reply:
@viktormalik, just to clarify: the `opts.attach_btf_obj_fd` cleanup is required else we leak fd's for kfunc, kretprobe and iter probes. I agree, the existing `opts.attach_prog_fd ` close is not needed and I've no removed that. It all tests out fine locally but will need to wait for the CI to turn green.

Reply:
For my own knowledge, what normally closes this file descriptor (in other bpf contexts)?